src/client/features/view/components/menu/search.js

```diff
@@ -0,0 +1,96 @@
+const _ = require('lodash');
+
+//Apply hover styling to a collection of nodes
+function updateStyling(style, matchedNodes) {
+  matchedNodes.forEach(node => {
+    let isCompartment = node.data('class') === 'compartment';
+
+    //Collapse Expanded Targets
+    if (node.isExpanded() && !isCompartment) { node.collapse(); }
+
+    //Apply style to the parent
+    if (node.isChild() && node.parent().data('class') !== 'compartment') { node = node.parent(); }
+
+    applySearchStyle(node, style);
+  });
+}
+
+//Determine if a regex pattern is valid
+function validateRegex(pattern) {
+  var parts = pattern.split('/'),
+    regex = pattern,
+    options = "";
+  if (parts.length > 1) {
+    regex = parts[1];
+    options = parts[2];
+  }
+  try {
+    let regexObj = new RegExp(regex, options);
+    return regexObj;
+  }
+  catch (e) {
+    return null;
+  }
+}
+
+function applySearchStyle(eles, style) {
+  eles.style(style);
+}
+
+//Search for nodes that match an entered query
+function searchNodes(query, cy) {
+  const searchValue = query.target.value;
+  const isBlank = _.isString(searchValue) ? !!_.trim(searchValue) : false;
+  const isRegularExp = _.startsWith(searchValue, 'regex:') && validateRegex(searchValue.substring(6));
+  const isExact = _.startsWith(searchValue, 'exact:');
+
+  let visibleNodes = Array.prototype.slice.call(cy.nodes(), 0);
+
+  //Add children of nodes to nodes list
+  const allChildNodes = visibleNodes.map(node => node.data('compoundCollapse.collapsedCollection'));
+  const validChildNodes = _.compact(_.flattenDeep(allChildNodes.map(collection => collection ? Array.prototype.slice.call(collection, 0) : null)));
+  const nodes = _.union(visibleNodes, validChildNodes);
```

Comment:
Generally for cytoscape collections I would prefer you use the cytoscape versions of these functions.

http://js.cytoscape.org/#eles.slice

http://js.cytoscape.org/#collection/building--filtering

Reply:
I have changed the implementation to use cytoscape eles.union()